src/Dialect/ONNX/Transforms/ConstProp.cpp

```diff
@@ -134,6 +145,46 @@ Value ConstZeroTensor(
           type, rewriter.getZeroAttr(type.getElementType())));
 }
 
+template <typename GetFPConstFunc =
+              std::function<APFloat(const llvm::fltSemantics &, bool)>,
+    typename GetIntConstFunc = std::function<APInt(unsigned)>>
+Value GetClipConstantOfType(PatternRewriter &rewriter, ShapedType type,
+    Value value, GetFPConstFunc fpConstantFunc, bool isNegative,
+    GetIntConstFunc intConstantFunc) {
```

Comment:
Since `value` is None in this case and it is used to get location only, how about not passing it in to avoid confusion? We can pass in a location instead.

e.g.
```c
Value GetClipConstantOfType(PatternRewriter &rewriter,  Location loc,
    ShapedType type, GetFPConstFunc fpConstantFunc, bool isNegative,
    GetIntConstFunc intConstantFunc) {
```

Then you can use OnnxBuilder to create a constant:
```c
OnnxBuilder create(rewriter, loc);
return create.constant(DenseElementsAttr::get(RankedTensorType::get({}, elemType), llvm::ArrayRef(fpValue)));
```

Also `min`, `max` in Clip are scalar, we can use a scalar type for the constant to make it consistent with the other case where `value` is not None.